itest/lnd_multi-hop_test.go

```diff
@@ -1838,3 +1838,165 @@ func createThreeHopNetwork(ht *lntest.HarnessTest,
+		ht, alice, bob, false, c, zeroConf,
+	)
+
+	// With the network active, we'll now add a new hodl invoice at Carol's
+	// end. Make sure the cltv expiry delta is large enough, otherwise Bob
+	// won't send out the outgoing htlc.
+	const invoiceAmt = 100000
+	var preimage lntypes.Preimage
+	copy(preimage[:], ht.Random32Bytes())
+	payHash := preimage.Hash()
+	invoiceReq := &invoicesrpc.AddHoldInvoiceRequest{
+		Value:      invoiceAmt,
+		CltvExpiry: finalCltvDelta,
+		Hash:       payHash[:],
+	}
+	eveInvoice := carol.RPC.AddHoldInvoice(invoiceReq)
+
+	// Subscribe the invoice.
+	stream := carol.RPC.SubscribeSingleInvoice(payHash[:])
+
+	// Now that we've created the invoice, we'll send a single payment from
+	// Alice to Carol. We won't wait for the response however, as Carol
+	// will not immediately settle the payment.
+	req := &routerrpc.SendPaymentRequest{
+		PaymentRequest: eveInvoice.PaymentRequest,
+		TimeoutSeconds: 60,
+		FeeLimitMsat:   noFeeLimitMsat,
+	}
+	alice.RPC.SendPayment(req)
+
+	// At this point, all 3 nodes should now have an active channel with
+	// the created HTLC pending on all of them.
+	htlc := ht.AssertHLTCActive(alice, aliceChanPoint, payHash[:])
+	ht.AssertHLTCActive(bob, bobChanPoint, payHash[:])
+	ht.AssertHLTCActive(carol, bobChanPoint, payHash[:])
+
+	// We let Alice go offline so the link between Alice and Bob is broken.
+	restartAlice := ht.SuspendNode(alice)
+
+	// Wait for carol to mark invoice as accepted. There is a small gap to
+	// bridge between adding the htlc to the channel and executing the exit
+	// hop logic.
+	ht.AssertInvoiceState(stream, lnrpc.Invoice_ACCEPTED)
+
+	// Carol now settles the invoice, which means Bob now has the preimage.
+	carol.RPC.SettleInvoice(preimage[:])
+
+	// Since Bob knows the preimage, the HTLC on the link in Bob->Carol
+	// should be settled.
+	//
+	// TODO(yy): We may get a closed circuit error here, need to
+	// investigate and fix.
+	// ht.AssertHLTCNotActive(bob, bobChanPoint, payHash[:])
+
+	// Stop Bob to avoid his contract court from force closing the channel
+	// and settling the HTLC onchain via the "normal" path.
+	restartBob := ht.SuspendNode(bob)
+
+	// Increase the fee estimate so that the following force close tx will
+	// be cpfp'ed.
+	ht.SetFeeEstimate(30000)
+
+	// Get the current height to compute number of blocks to mine to
+	// trigger the force close from Alice.
+	_, height := ht.Miner.GetBestBlock()
+
+	// We'll now mine enough blocks to trigger Alice's broadcast of her
+	// commitment transaction due to the fact that the HTLC is about to
+	// timeout. With the default outgoing broadcast delta of zero, this
+	// will be the same height as the htlc expiry height.
+	numBlocks := htlc.ExpirationHeight - uint32(height) -
+		lncfg.DefaultOutgoingBroadcastDelta
+	ht.MineBlocks(numBlocks)
+
+	// Restart Alice again. Once she finishes syncing the channel state,
+	// she should force close the channel Alice->Bob.
+	require.NoError(ht, restartAlice())
```

Comment:
Why should Alice force close? We should mine juuust enogh blocks for the outgoing HTLC to expire (CLTV delta is remaining at this point), with Alice actually getting the preimage from Bob at the very end.